Mage.Sets/src/mage/cards/p/ProgenitorsIcon.java

```diff
@@ -0,0 +1,148 @@
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+        ProgenitorsIconWatcher.addPlayer(source.getControllerId(), subType, game);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public ProgenitorsIconAsThoughEffect copy() {
+        return new ProgenitorsIconAsThoughEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+
+        if (!ProgenitorsIconWatcher.checkPlayerCast(affectedControllerId, subType, game)) {
+            discard();
+            return false;
+        }
+        Card card = game.getCard(sourceId);
+        if (card == null){
+            return false;
+        }
+
+        return card.getSubtype().contains(subType);
+    }
+}
+
+class ProgenitorsIconWatcher extends Watcher {
+
+    // Handles multiple instances of this same card choosing the same or different subtypes for the same player
+    private final Map<UUID, ArrayList<SubType>> playerSubTypesChosen = new HashMap<>();
+
+    public ProgenitorsIconWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.SPELL_CAST){
+            UUID playerID = event.getPlayerId();
+            Spell spell = game.getSpellOrLKIStack(event.getSourceId());
+            ArrayList<SubType> chosenSubTypes = playerSubTypesChosen.get(playerID);
+            // Remove all matching subtypes. If you tap two Progenitor's Icons for flash applying to Human and Goblin
+            // subtypes, then cast a Human Goblin with flash, both instances of Progenitor's Icon abilities get used.
+            for (SubType subType : spell.getSubtype()){
+                chosenSubTypes.remove(subType);
+            }
+        }
+    }
+
+    public static void addPlayer(UUID playerId, SubType subtype, Game game) {
+        Map<UUID, ArrayList<SubType>> playerSubTypesChosen =
+                game.getState().getWatcher(ProgenitorsIconWatcher.class).playerSubTypesChosen;
+        if (!playerSubTypesChosen.containsKey(playerId)){
+            playerSubTypesChosen.put(playerId, new ArrayList<>());
+        }
+        playerSubTypesChosen.get(playerId).add(subtype);
+    }
+
+    public static boolean checkPlayerCast(UUID playerId, SubType subtype, Game game) {
+        Map<UUID, ArrayList<SubType>> playerSubTypesChosen =
+                game.getState().getWatcher(ProgenitorsIconWatcher.class).playerSubTypesChosen;
+        if (!playerSubTypesChosen.containsKey(playerId)){
+            return false;
+        }
+        return playerSubTypesChosen.get(playerId).contains(subtype);
```

Comment:
similarly you could use

```java
playerSubtypesChosen.getOrDefault(playerId, Collections.emptySet()).contains(subtype)
```
to avoid checking `containsKey`

(whether you think this is cleaner or not is a style decision)